capabilityreceiver/pkg/capabilityhandler/includedfields.go

```diff
@@ -0,0 +1,82 @@
+package capabilityhandler
+
+// RunIncludedFieldsChecks stores whether each field in capability statement is populated or not populated
+func RunIncludedFieldsChecks(capInt map[string]interface{}) map[string]bool {
+	if capInt == nil {
+		return nil
+	}
+	includedFields := make(map[string]bool)
+
+	fieldsList := [][]string{
+		{"url"},
+		{"version"},
+		{"name"},
+		{"title"},
+		{"status"},
+		{"experimental"},
+		{"date"},
+		{"publisher"},
+		{"contact"},
+		{"contact", "name"},
+		{"contact", "telecom"},
+		{"description"},
+		{"requirements"},
+		{"useContext"},
+		{"jurisdiction"},
+		{"purpose"},
+		{"copyright"},
+		{"kind"},
+		{"instantiates"},
+		{"imports"},
+		{"software", "name"},
+		{"software", "version"},
+		{"software", "releaseDate"},
+		{"implementation", "description"},
+		{"implementation", "url"},
+		{"implementation", "custodian"},
+		{"fhirVersion"},
+		{"format"},
+		{"patchFormat"},
+		{"acceptUnknown"},
+		{"implementationGuide"},
+		{"profile"},
+	}
+
+	for _, fieldNames := range fieldsList {
+		var stringIndex string
+		if len(fieldNames) == 2 {
```

Comment:
This should be dynamic so that it can handle any length of names ie: any nest depth.

Reply:
Good point, I fixed this so making the stringIndex can handle any length of names now